Pomodoro/app/src/main/java/no/uia/ikt205/pomodoro/MainActivity.kt

```diff
@@ -4,70 +4,176 @@ import androidx.appcompat.app.AppCompatActivity
-            if (coutdownDisplay.text == millisecondsToDescriptiveTime(0)){
-                startCountDown(it)
+        coutdownDisplay = findViewById<TextView>(R.id.countDownView)
+    }
+
+    fun startCountDown(v: View) {
+
+        // If we try and start a timer we check if either timer is already on
+        // If either the normal countdown or pause timer are on we restart first
+        // and then the user can start their timer
+
+        if (isTimerOn or isPauseOn) {
+            timer.cancel()
+            isTimerOn = false
+            isPauseOn = false
+
+        } else {
+            // If no timers are on we set the countdown variable to true
+            isTimerOn = !isTimerOn
+
+            timer = object : CountDownTimer(timeToCountDownInMs, timeTicks) {
+                override fun onFinish() {
+                    // when it is finished we set our countdown variable to false
+                    isTimerOn = false
+                    Toast.makeText(this@MainActivity, "Arbeidsøkt er ferdig", Toast.LENGTH_SHORT)
+                        .show()
+                    if (!isPauseOn) {
+                        // we then start the pause timer
+                        Toast.makeText(this@MainActivity, "Time for a break", Toast.LENGTH_SHORT).show()
+                        startPauseTimer(v)
+
+                    }
+                }
+
+                override fun onTick(millisUntilFinished: Long) {
+                    updateCountDownDisplay(millisUntilFinished)
+                }
             }
-       }
-       coutdownDisplay = findViewById<TextView>(R.id.countDownView)
+            timer.start()
+        }
 
     }
 
-    fun startCountDown(v: View){
-        timer = object : CountDownTimer(timeToCountDownInMs,timeTicks) {
+    fun startPauseTimer(v: View) {
+        // We now flip isPauseOn to true, as the pause timer only starts after a work we don't have to worry
+        // about conflicts with pressing start again
+
+        isPauseOn = !isPauseOn
+        timer = object : CountDownTimer(pauseTimer, timeTicks) {
             override fun onFinish() {
-                Toast.makeText(this@MainActivity,"Arbeidsøkt er ferdig", Toast.LENGTH_SHORT).show()
+                // If our repetition amounts are higher than 1 we start the repetition cycle
+                // We decrement our repAmount by 1 and start the timer all over again
+                // Once the repAmount is set to 0 it no longer repeats
+                if (repAmount > 0){
+                    Toast.makeText(this@MainActivity, "Time to start working again", Toast.LENGTH_SHORT).show()
+                    isPauseOn = false
+                    repAmount--
+                    repeatAmountText.setText(repAmount.toString())
+                    startCountDown(v)
+                } else {
+                    isPauseOn = false
+                    Toast.makeText(this@MainActivity, "Timer is done", Toast.LENGTH_SHORT).show()
+
+                }
             }
```

Comment:
Good writen code. Seems well functional. Helps with the comments to understand what you are trying to make as well. Good job.